File: src/main/java/am/adrian/dungeonkeeper/model/gameobject/GameObject.java
package am.adrian.dungeonkeeper.model.gameobject;

import am.adrian.dungeonkeeper.model.coords.Coords;
import am.adrian.dungeonkeeper.model.gameobject.trait.CanCollide;

public interface GameObject extends CanCollide {

    Coords getCoords();

    String getTexture();
}

File: src/main/kotlin/am/adrian/dungeonkeeper/model/gameobject/GameObject.kt
package am.adrian.dungeonkeeper.model.gameobject

import am.adrian.dungeonkeeper.model.coords.Coords
import am.adrian.dungeonkeeper.model.gameobject.trait.CanCollide

interface GameObject : CanCollide {

    fun getCoords(): Coords

    fun getTexture(): String
}
